README.md

```diff
@@ -1,189 +1,337 @@
+var s = region(ispace(int1d, 3), complex64)
 ```
+
+The plan region always takes the following form, with fieldspace `fft.plan` (see
+`fft.rg` for description of plan fieldspace):
+
+```lua
 var p = region(ispace(int1d, 1), fft1d.plan)
-fft1d.make_plan(r, s, p)
 ```
 
-**Important:** `make_plan` overwites the input and output regions `r`
-and `s`. This is mandated by FFTW, which Regent FFT uses on CPUs. In
-order to avoid overwriting data, the user must either initialize the
-plan prior to loading the regions with data, or else must create a
-temporary region (of the same size and layout as the real one) for use
-in initialization.
+`make_plan` is a `__demand(__inline)` task. This means that if the user wants it
+to execute it in a separate task, they must wrap the task themselves.
 
-Note that, like all non-distributed mode tasks, `make_plan` is a
-`__demand(__inline)` task. This means that if the user wants it to
-execute it in a separate task, they must wrap the task themselves.
+> [!IMPORTANT]
+>
+> `make_plan` overwrites the input and output regions `r` and `s`. This is
+> mandated by FFTW, which Regent FFT uses on CPUs. In order to avoid overwriting
+> data, the user must either initialize the plan prior to loading the regions
+> with data, or else must create a temporary region (of the same size and layout
+> as the real one) for use in initialization.
 
-To initialize in distributed mode:
+#### 3. Execute Plan
 
+Next, we execute the plan. This takes the same 3 regions as mentioned above.
+
+```lua
+fft1d.execute_plan_task(r, s, p)
 ```
-var n = fft1d.get_num_nodes()
-var p = region(ispace(int1d, n), fft1d.plan)
-var p_part = partition(equal, p, ispace(int1d, n))
-fft1d.make_plan_distrib(r, r_part, s, s_part, p, p_part)
-```
 
-Note the use of `get_num_nodes` to determine the size of the `p`
-region and partition. The task `make_plan_distrib` is a
-`__demand(__inline)` task that internally performs an index launch
-over the machine to initialize `p`.
+Note that `execute_plan` is a `__demand(__inline)` task (similar to `make_plan`
+above). The task `execute_plan_task` is simply a wrapper around `execute_plan`
+for convenience, to avoid needing to define this explicitly.
+
+> [!IMPORTANT]
+>
+> Because `execute_plan` is a `__demand(__inline)` task, it will never execute
+> on the GPU (unless the parent task is running on the GPU). Therefore, in most
+> cases, it is necessary to use `execute_plan_task` if one wants to use the GPU.
 
-**Important:** like `make_plan`, `make_plan_distrib` overwrites the
-input and output regions `r` and `s`.
+#### 4. Destroy Plan
 
-**Important:** in order for the distributed API to work correctly, it
-is essential that each task in the index launch inside of
-`make_plan_distrib` is mapped onto a separate node. This ensures that
-when the region `p` is used later, there is a plan for every node in
-the machine.
+When a plan is no longer needed, it can be destroyed:
+
+```lua
+fft1d.destroy_plan(p)
```

Comment:
I think we should be consistent and call the same version for each usage of the API we're showing. So `make_plan`, `execute_plan` and `destroy_plan`; or `make_plan_task`, `execute_plan_task` and `destroy_plan_task`.

We should probably make a point about being consistent about what you do: because the processor that runs `make_plan` should probably be the same processor you `execute_plan` and `destroy_plan`.

Reply:
In the readme, I now call `make_plan`, `execute_plan` and `destroy_plan` for the regular version and `make_plan_distrib`, `execute_plan_distrib` and `destroy_plan_distrib` for the distributed version.